datafusion/core/src/execution/context.rs

```diff
@@ -1923,59 +1846,15 @@ impl TaskContext {
         aggregate_functions: HashMap<String, Arc<AggregateUDF>>,
         runtime: Arc<RuntimeEnv>,
     ) -> Self {
-        let session_config = if task_props.is_empty() {
-            SessionConfig::new()
-        } else {
-            SessionConfig::new()
-                .with_batch_size(task_props.get(OPT_BATCH_SIZE).unwrap().parse().unwrap())
-                .with_target_partitions(
-                    task_props
-                        .get(OPT_TARGET_PARTITIONS)
-                        .unwrap()
-                        .parse()
-                        .unwrap(),
-                )
-                .with_repartition_joins(
-                    task_props
-                        .get(OPT_REPARTITION_JOINS)
-                        .unwrap()
-                        .parse()
-                        .unwrap(),
-                )
-                .with_repartition_aggregations(
-                    task_props
-                        .get(OPT_REPARTITION_AGGREGATIONS)
-                        .unwrap()
-                        .parse()
-                        .unwrap(),
-                )
-                .with_repartition_windows(
-                    task_props
-                        .get(OPT_REPARTITION_WINDOWS)
-                        .unwrap()
-                        .parse()
-                        .unwrap(),
-                )
-                .with_parquet_pruning(
-                    task_props
-                        .get(OPT_PARQUET_ENABLE_PRUNING)
-                        .unwrap()
-                        .parse()
-                        .unwrap(),
-                )
-                .with_collect_statistics(
-                    task_props
-                        .get(OPT_COLLECT_STATISTICS)
-                        .unwrap()
-                        .parse()
-                        .unwrap(),
-                )
-        };
+        let mut config = ConfigOptions::new();
+        for (k, v) in task_props {
+            let _ = config.set(&k, &v);
```

Comment:
This shows deserializing from a HashMap

Reply:
related to https://github.com/apache/arrow-datafusion/pull/4758